Library/Homebrew/version.rb

```diff
@@ -379,6 +379,21 @@ def self._parse(spec, detected_from_url:)
   end
   private_class_method :_parse
 
+  DOT_OPTIONAL = /(?:\d+(?:\.\d+)*)/.source.freeze
+  private_constant :DOT_OPTIONAL
+
+  DOT_REQUIRED = /(?:\d+(?:\.\d+)+)/.source.freeze
+  private_constant :DOT_REQUIRED
```

Comment:
The salient difference in these regexes is in the number of required parts (a minimum of 1 or a minimum of 2). To me, the `DOT_OPTIONAL` and `DOT_REQUIRED` names focus on the presence of a dot and don't express that these regexes are intended to match the numeric part of versions (e.g., `1`, `1.2`, `1.2.3`, etc.).

However, I'm not quite sure what we could use for better names that wouldn't be overly verbose. Maybe `VER_PARTS_OPT` and `VER_PARTS_REQ`? Does anyone have other suggestions?

Reply:
I think `DOT_OPTIONAL` is the most succinct in this context as I think most developers understand that a dot in a version string means e.g. `1.0`

As this constant is inside the `Version` class in `version.rb`, it's probably a bit redundant to add `VER_` to the constant name